comp/metadata/clusteragent/impl/cluster_agent.go

```diff
@@ -127,19 +132,26 @@ func (dca *datadogclusteragent) getPayload() marshaler.JSONMarshaler {
 	}
 }
 
+func scrub(s string) string {
+	// Errors come from internal use of a Reader interface. Since we are reading from a buffer, no errors
+	// are possible.
+	scrubString, _ := scrubber.ScrubString(s)
+	return scrubString
+}
+
 func (dca *datadogclusteragent) initMetadata() {
 	tool := "undefined"
 	toolVersion := ""
 	installerVersion := ""
 
-	install, err := installinfo.Get(dca.conf)
+	install, err := installinfoGet(dca.conf)
```

Comment:
I think `installinfo.Get` is only used for metadata, we could update that method to return scrubbed data instead of all the metadata provider. WDYT ?

Also, should we scrub everything returned, not just the tool ? Just to be on the safe side.

Reply:
Done, scrubbing is now being done in `pkg/util/installinfo` instead. Keeping the tests modified for each call-site in order to validate that they do the right thing.